### PR TITLE
Allow spaces and `-` in secret identifiers

### PR DESCRIPTION
**Pull request by Oleksiivanov**
Corrected secrets.js regex pattern;

Links to issue : https://camunda.slack.com/archives/C034N0E5YCX/p1696966815142039

### CODE REVIEW

File: test/camunda-cloud/secrets.spec.js [hunk 1]
```diff
@@ -51,6 +51,18 @@ const valid = [
...
       </bpmn:intermediateCatchEvent>
     `))
+  },
+  {
+    name: 'property with valid value with dashes',
+    moddleElement: createModdle(createProcess(`
+      <bpmn:intermediateCatchEvent id="IntermediateCatchEvent_1">
+        <bpmn:extensionElements>
+          <zeebe:properties>
+            <zeebe:property name="bar" value="{{secrets.FOO-FOO}}" />
```
Comment (nikku) [inline test/camunda-cloud/secrets.spec.js:61]: Does `{{ secrets.FOO }}` also match? If so, let's verify that pattern works, too, through a test case.

File: test/camunda-cloud/secrets.spec.js [hunk 2]
```diff
@@ -51,6 +51,30 @@ const valid = [
...
       </bpmn:intermediateCatchEvent>
     `))
+  },
+  {
+    name: 'property with valid value with dashes',
+    moddleElement: createModdle(createProcess(`
+      <bpmn:intermediateCatchEvent id="IntermediateCatchEvent_1">
+        <bpmn:extensionElements>
+          <zeebe:properties>
+            <zeebe:property name="bar" value="{{secrets.FOO}}" />
```
Comment (nikku) [inline test/camunda-cloud/secrets.spec.js:61]:
```suggestion
            <zeebe:property name="bar" value="{{secrets.FOO-BAR}}" />
```

File: test/camunda-cloud/secrets.spec.js [hunk 3]
```diff
@@ -51,6 +51,30 @@ const valid = [
...
+      </bpmn:intermediateCatchEvent>
+    `))
+  },
+  {
+    name: 'property with valid value with spaces',
+    moddleElement: createModdle(createProcess(`
+      <bpmn:intermediateCatchEvent id="IntermediateCatchEvent_1">
+        <bpmn:extensionElements>
+          <zeebe:properties>
+            <zeebe:property name="bar" value="{{ secrets.FOO-FOO }}" />
```
Comment (nikku) [inline test/camunda-cloud/secrets.spec.js:73]:
```suggestion
            <zeebe:property name="bar" value="{{  secrets.FOO_BAR   }}" />
```

File: test/camunda-cloud/secrets.spec.js (hunk 1 above)
Comment (nikku) [reply test/camunda-cloud/secrets.spec.js]: Thanks. Added some follow-ups, potentially making the tests more robust.

File: rules/camunda-cloud/secrets.js [hunk 4]
```diff
@@ -115,5 +115,5 @@ function isValidSecret(value) {
   return !value
     || !isString(value)
     || !value.includes('secrets.')
-    || /{{secrets\.[a-zA-Z0-9_]+}}/.test(value);
+    || /{{\s*secrets\.(\S+?\s*)}}/.test(value);
```
Comment (nikku) [inline rules/camunda-cloud/secrets.js:118]:
```suggestion
    || /{{\s*secrets\.[\S]+\s*}}/.test(value);
```

This matches:

* Any amount of whitespace
* `secrets.`
* Any amount of characters, not including whitespace
* Any amount of whitespace

If we rather want to be on the safer side, we want to restrict to word only chars (cf. original rule) + `-`:

```suggestion
    || /{{\s*secrets\.[\w-]+\s*}}/.test(value);
```

**Comment by nikku** [comment]
Great stuff, thanks for the contribution.

**Comment by nikku** [comment]
Released with `v2.9.1`.

**Comment by philippfromme** [comment]
@Oleksiivanov Does this now reflect the regular expression used for connector runtime?

**Pull request merged**